code/modules/admin/player_panel/actions/punish.dm

```diff
@@ -186,10 +186,12 @@
 			return
 		if("No")
 			new_name = random_name(target_mob.gender)
+			GLOB.data_core.manifest_modify(new_name, WEAKREF(target_mob))
 		if("Yes")
 			var/raw_name = input(user, "Choose the new name:", "Name Input")  as text|null
 			if(!isnull(raw_name)) // Check to ensure that the user entered text (rather than cancel.)
 				new_name = reject_bad_name(raw_name)
+				GLOB.data_core.manifest_modify(new_name, WEAKREF(target_mob))
```

Comment:
Probably want both of these added procs down right after target_mob.change_real_name() so you don't duplicate the code and we guarantee the name was changed.